utils/docker-ce/files/dockerd.init

```diff
@@ -43,10 +43,39 @@ process_config() {
 	json_dump > "$DOCKERD_CONF"
 }
 
+process_ac_allowed() {
+	local cfg="$1"
+
+	local iface
+
+	config_get iface "$cfg" iface
+	iptables -A DOCKER-ACCESS-CONTROL -i "$iface" -o docker0 -j RETURN
+
+}
+
+process_ac() {
+	[ -f /etc/config/dockerd ] || {
+		return 0
+	}
+
+	# Add access control
+	iptables -N DOCKER-ACCESS-CONTROL
+
+	config_foreach process_ac_allowed access_control
+
+	iptables -A DOCKER-ACCESS-CONTROL -m conntrack --ctstate ESTABLISHED,RELATED -o docker0 -j RETURN
+	iptables -A DOCKER-ACCESS-CONTROL -m conntrack --ctstate NEW,INVALID -o docker0 -j DROP
+	iptables -A DOCKER-ACCESS-CONTROL -j RETURN
+
+	iptables -N DOCKER-USER
+	iptables -I DOCKER-USER -j DOCKER-ACCESS-CONTROL
+}
+
 start_service() {
 	local nofile=$(cat /proc/sys/fs/nr_open)
 
 	process_config
+	process_ac
```

Comment:
Logically, AC is defined in the configuration, so maybe we should call this function in `process_config`?
We could also rely on the `if [` at line 17 if we call it somewhere  like line 33 then remove the duplicate at line 57 that way.

Reply:
Sounds good. Will do update my P/R.